irods/message/__init__.py

```diff
@@ -50,37 +52,59 @@ class XML_Parser_Type(MyIntEnum):
 #   - the server version with which we're communicating
 #   - which of the choices of parser (STANDARD_XML or QUASI_XML) we're using
 
 _thrlocal = threading.local()
 
-# The packStruct message parser defaults to STANDARD_XML but you can override it by setting the
-# environment variable PYTHON_IRODSCLIENT_QUASI_XML_DEFAULT_SERVER_VERSION.
-# If this variable is defined in the environment, its value:
-#    * must be tuple-like and of a commensurate or compatible version with the connected server.
-#    * should be <= 4,2,8 for older servers wherein backticks were incorrectly encoded as '&apos;'
-#    * can be () or >=4,2,9 for newer servers to allow a flexible character set within iRODS protocol.
-#    * empty or not defined, in which case the standard XML protocol is used.
+# The packStruct message parser defaults to STANDARD_XML but we can override it by setting the
+# environment variable PYTHON_IRODSCLIENT_DEFAULT_XML to either 'SECURE_XML' or 'QUASI_XML'.
+# If QUASI_XML is selected, the environment variable PYTHON_IRODSCLIENT_QUASI_XML_SERVER_VERSION
+# may also be set to a tuple "X,Y,Z" to inform the client of the connected iRODS server version.
+# If we set a value for the version, it can be either:
+#    * 4,2,8 to work with that server version and older ones which incorrectly encoded back-ticks as '&apos;'
+#    * an empty tuple "()" or something >= 4,2,9 to work with newer servers to allow a flexible character
+#      set within iRODS protocol.
 
-_Quasi_Xml_Default_Server_Version = _qxml_server_version('PYTHON_IRODSCLIENT_QUASI_XML_DEFAULT_SERVER_VERSION')
+class BadXMLSpec(RuntimeError): pass
 
-_current_XML_parser = XML_Parser_Type.QUASI_XML if _Quasi_Xml_Default_Server_Version is not None \
-                 else XML_Parser_Type.STANDARD_XML
+_Quasi_Xml_Server_Version = _qxml_server_version('PYTHON_IRODSCLIENT_QUASI_XML_SERVER_VERSION')
+if _Quasi_Xml_Server_Version is None:  # unspecified in environment yields empty tuple ()
+    raise BadXMLSpec('Must properly specify a server version to use QUASI_XML')
 
-def current_XML_parser():
-    return getattr(_thrlocal,'xml_type',_current_XML_parser)
+_XML_strings = { k:v for k,v in vars(XML_Parser_Type).items() if k.endswith('_XML')}
+
+
+_default_XML = os.environ.get('PYTHON_IRODSCLIENT_DEFAULT_XML','')
+if not _default_XML:
+    _default_XML = XML_Parser_Type.STANDARD_XML
+else:
+    try:
+        _default_XML = _XML_strings[_default_XML]
+    except KeyError:
+        raise BadXMLSpec('XML parser type not recognized')
+
+
+def current_XML_parser(get_module = False):
+    d = getattr(_thrlocal,'xml_type',_default_XML)
+    return d if not get_module else _XML_parsers[d]
+
+def default_XML_parser(get_module = False):
+    d = _default_XML
+    return d if not get_module else _XML_parsers[d]
 
 _XML_parsers = {
     XML_Parser_Type.STANDARD_XML : ET_xml,
-    XML_Parser_Type.QUASI_XML : ET_quasixml
+    XML_Parser_Type.QUASI_XML : ET_quasixml,
```

Comment:
consider `ET_quasi_xml`?    for consistency?

don't worry about it if it's already shipped and this would be a change that affects anything.

Reply:
Sure, that can be changed with no likely impact to anyone.